complete/src/main/java/hello/Application.java

```diff
@@ -13,7 +13,10 @@
 public class Application extends AsyncConfigurerSupport {
 
     public static void main(String[] args) {
-        SpringApplication.run(Application.class, args);
+        SpringApplication.run(Application.class, args).close();
+
+        // Close application
+        System.exit(0);
     }
```

Comment:
It the `ExecutorService` is defined as a `@Bean` instead of using `AsyncConfigurerSupport` then by closing the application context the `ExecutorService` shuts down.